src/config.js

```diff
@@ -0,0 +1,179 @@
+/* eslint-env mocha */
+'use strict'
+
+const expect = require('chai').expect
+
+module.exports = (common) => {
+  describe('.config', () => {
+    let ipfs
+
+    before((done) => {
+      common.setup((err, _ipfs) => {
+        expect(err).to.not.exist
+        ipfs = _ipfs
+        done()
+      })
+    })
+
+    after((done) => {
+      common.teardown(done)
+    })
+
+    describe('callback API', () => {
+      describe('.get', () => {
+        it('retrieve the whole config', (done) => {
+          ipfs.config.get((err, config) => {
+            expect(err).to.not.exist
+            expect(config).to.exist
+            done()
+          })
+        })
+
+        it('retrieve a value through a key', (done) => {
+          ipfs.config.get('Identity', (err, identity) => {
+            expect(err).to.not.exist
+            expect(identity).to.exist
+            done()
+          })
+        })
+
+        it('retrieve a value through a nested key', (done) => {
+          ipfs.config.get('Addresses.Swarm', (err, swarmAddrs) => {
```

Comment:
I don't think that is required as in: will ever be used, since a value used by a component of the protocol will always look for a key that belongs to it and not to a partial set of the value of a key.